Build fails on Manjaro/Arch (and other systems using OpenCV with OpenEXR 3.2+) if libdeflate isn't explicitly linked

### Bug Description

When I compile Tahoma2D, following the Linux build instructions either as is or with a modified PKGBUILD, it always stops around 62% after it says it finished building something, e.g. `toonzqt` or `tnzstdfx`, with the error `make: *** [Makefile:136: all] Error 2`. If I try compiling again a few times, somehow it seems to get past that part and I can even get to 100% (after it says it finished building the `Tahoma2D` executable) but still have this error. Adding `LANG=C` and `VERBOSE=1` doesn't give any more helpful/meaningful output.

This has been a problem since around August or September, and I don't know what changed.

Edit: Possibly something to do with `openexr` updating from 3.1.11 to 3.2.0 and installing `libdeflate` as a dependency (has a linker error, usually further up around 42% or 43%, about undefined references in `libimage` to `libdeflate`).

### Steps to reproduce

1. Start building from command line, following instructions
2. Error occurs

### Relevant logs/screenshots

```
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_decompressor'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_compressor'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_decompress'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress_bound'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_compressor'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_decompressor'
collect2: error: ld returned 1 exit status
```

### Software Version

Other (Specify below)

### Software Version Information

Commit 919d53815 (or whichever is latest at the time, since I try again with each commit)

### Operating System

Manjaro with Linux kernel 6.5.7

### Graphics Card

_No response_

### Drawing Tablet

_No response_

From the log:

```
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_decompressor'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_compressor'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_decompress'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress_bound'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_compressor'
/usr/bin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_decompressor'
collect2: error: ld returned 1 exit status
```

Hello, I'm the maintainer of the AUR pkgbuilds of tahoma2d and tahoma2d-git, in addition to the libdeflate problem I have an issue with libwebp

With libwebp I have that problem since earlier this year, v is 1.3.2, its build option can be seen here https://gitlab.archlinux.org/archlinux/packaging/packages/libwebp/-/blob/main/PKGBUILD?ref_type=heads


```
[ 47%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/spectrumfield.cpp.o
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPIAppend'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPEncode'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_decompressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPIDecGetRGB'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_compressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_decompress'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPIDelete'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPInitDecBufferInternal'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureImportRGBA'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureFree'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPConfigInitInternal'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress_bound'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPINewDecoder'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPFreeDecBuffer'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureInitInternal'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureImportRGB'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_compressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_decompressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPValidateConfig'
collect2: error: ld returned 1 exit status
make[2]: *** [tcleanupper/CMakeFiles/tdcleanup.dir/build.make:120: bin/tdcleanup] Error 1
make[1]: *** [CMakeFiles/Makefile2:843: tcleanupper/CMakeFiles/tdcleanup.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 47%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/spreadsheetviewer.cpp.o
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-#warnings' may have been intended to silence earlier diagnostics
[ 47%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_hard_light.cpp.o
[ 47%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_hard_mix.cpp.o
[ 47%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/strokesdata.cpp.o
[ 47%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/studiopaletteviewer.cpp.o
[ 47%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_lighten.cpp.o
[ 47%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_lighter_color.cpp.o
[ 47%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/styleeditor.cpp.o
[ 48%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_linear_burn.cpp.o
[ 48%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_linear_dodge.cpp.o
[ 48%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_linear_light.cpp.o
[ 48%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/styleindexlineedit.cpp.o
[ 48%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_multiply.cpp.o
[ 48%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_over.cpp.o
[ 48%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/swatchviewer.cpp.o
[ 49%] Linking CXX executable ../bin/tdconverter
[ 49%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_overlay.cpp.o
[ 49%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_pin_light.cpp.o
[ 50%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/tabbar.cpp.o
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_screen.cpp.o
[ 50%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/tdockwindows.cpp.o
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_soft_light.cpp.o
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_subtract.cpp.o
[ 50%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/tonecurvefield.cpp.o
[ 50%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/treemodel.cpp.o
[ 50%] Building CXX object toonzqt/CMakeFiles/toonzqt.dir/tselectionhandle.cpp.o
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blend_vivid_light.cpp.o
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPIAppend'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPEncode'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_decompressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPIDecGetRGB'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_compressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_decompress'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPIDelete'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPInitDecBufferInternal'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureImportRGBA'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureFree'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPConfigInitInternal'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress_bound'
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_blur.cpp.o
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_zlib_compress'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPINewDecoder'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPFreeDecBuffer'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureInitInternal'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPPictureImportRGB'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_free_compressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `libdeflate_alloc_decompressor'
/usr/sbin/ld: ../lib/tahoma2d/libimage.so: undefined reference to `WebPValidateConfig'
collect2: error: ld returned 1 exit status
make[2]: *** [tconverter/CMakeFiles/tdconverter.dir/build.make:119: bin/tdconverter] Error 1
make[1]: *** [CMakeFiles/Makefile2:908: tconverter/CMakeFiles/tdconverter.dir/all] Error 2
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_channel_selector.cpp.o
[ 50%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/ino_common.cpp.o
```

--- a/ci-scripts/linux/tahoma-install.sh
+++ b/ci-scripts/linux/tahoma-install.sh
@@ -6,6 +6,7 @@ sudo apt-get install -y cmake liblzo2-dev liblz4-dev libfreetype6-dev libpng-dev
 sudo apt-get install -y nasm yasm libgnutls28-dev libunistring-dev libass-dev libbluray-dev libmp3lame-dev libopus-dev libsnappy-dev libtheora-dev libvorbis-dev libvpx-dev libwebp-dev libxml2-dev libfontconfig1-dev libfreetype6-dev libopencore-amrnb-dev libopencore-amrwb-dev libspeex-dev libsoxr-dev libopenjp2-7-dev
 sudo apt-get install -y python3-pip
 sudo apt-get install -y build-essential libgirepository1.0-dev autotools-dev intltool gettext libtool patchelf autopoint libusb-1.0-0 libusb-1.0-0-dev
+sudo apt-get install -y libdeflate-dev
 
 pip3 install --upgrade pip
 pip3 install numpy

--- a/toonz/sources/CMakeLists.txt
+++ b/toonz/sources/CMakeLists.txt
@@ -541,6 +541,8 @@ elseif(BUILD_ENV_UNIXLIKE)
         find_library(GPHOTO2_PORT_LIB gphoto2_port)
         message("**************** gphoto2_port lib:" ${GPHOTO2_PORT_LIB})
     endif()
+	
+	find_library(DEFLATE_LIB deflate)
 endif()
 
 

--- a/toonz/sources/image/CMakeLists.txt
+++ b/toonz/sources/image/CMakeLists.txt
@@ -164,6 +164,7 @@ elseif(BUILD_ENV_UNIXLIKE)
     # Generic Unix
     set(EXTRA_LIBS
         ${TNZLIBS}
+        ${DEFLATE_LIB}
     )
 
     if(BUILD_TARGET_WIN)

